cli/command/system/prune.go

```diff
@@ -0,0 +1,144 @@
+package system
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/cli"
+	"github.com/docker/docker/cli/command"
+	units "github.com/docker/go-units"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+type pruneOptions struct {
+	force      bool
+	aggressive bool
+}
+
+// NewPruneCommand creates a new cobra.Command for `docker du`
+func NewPruneCommand(dockerCli *command.DockerCli) *cobra.Command {
+	var opts pruneOptions
+
+	cmd := &cobra.Command{
+		Use:   "prune [COMMAND]",
+		Short: "Delete unused data",
+		Args:  cli.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runPrune(dockerCli, true, true, true, opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&opts.force, "force", "f", false, "Do not prompt for confirmation")
+	flags.BoolVarP(&opts.aggressive, "aggressive", "a", false, "Delete all unused images not just dangling ones")
+
+	cmd.AddCommand(
+		&cobra.Command{
+			Use:   "containers",
+			Short: "Delete unused containers",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, true, false, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "volumes",
+			Short: "Delete unused volumes",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, true, false, opts)
+			},
+		},
+		&cobra.Command{
+			Use:   "images",
+			Short: "Delete unused images",
+			Args:  cli.NoArgs,
+			RunE: func(cmd *cobra.Command, args []string) error {
+				return runPrune(dockerCli, false, false, true, opts)
+			},
+		},
+	)
```

Comment:
I won't block on this point, but personally I don't see the value in having a single command removing everything:
* once we start adding filters it's going to be really tough to have a clean and sensible set of flags here
* I prefer to have "one way to do things", instead of two
* you can always run `docker volume prune; docker image prune; docker container prune`
* I think container prune is generally much safer than image prune, which is also much safer than volume prune. Doing them all at once kind of hides the danger in pruning volumes.

If we do keep a single place to prune everything I do think that just `docker system prune` would be better than having the subcommands.